src/CalcManager/Ratpack/conv.cpp

```diff
@@ -184,33 +164,19 @@ void _destroyrat(_In_ PRAT prat)
 //
 //    ARGUMENTS: size of number in 'digits'
 //
-//    RETURN: pointer to a number
+//    RETURN: a number
 //
-//    DESCRIPTION: allocates and zeros out number type.
+//    DESCRIPTION: initializes and zeros out number type.
 //
 //-----------------------------------------------------------------------------
 
-PNUMBER _createnum(_In_ uint32_t size)
+NUMBER _createnum(_In_ uint32_t size)
 
 {
-    PNUMBER pnumret = nullptr;
-    uint32_t cbAlloc;
-
-    // sizeof( MANTTYPE ) is the size of a 'digit'
-    if (SUCCEEDED(Calc_ULongAdd(size, 1, &cbAlloc)) && SUCCEEDED(Calc_ULongMult(cbAlloc, sizeof(MANTTYPE), &cbAlloc))
-        && SUCCEEDED(Calc_ULongAdd(cbAlloc, sizeof(NUMBER), &cbAlloc)))
-    {
-        pnumret = (PNUMBER)zmalloc(cbAlloc);
-        if (pnumret == nullptr)
-        {
-            throw(CALC_E_OUTOFMEMORY);
-        }
-    }
-    else
-    {
-        throw(CALC_E_INVALIDRANGE);
-    }
-    return (pnumret);
+    NUMBER num = NUMBER();
```

Comment:
This would create a default-constructed value, which will have `sign` equal 1: https://github.com/microsoft/calculator/blob/63e4a124ba08bc2ff0afd84164ea0fb9d05335cf/src/CalcManager/Ratpack/ratpak.h#L63

That's unlike the current/old code, which would initialise it to 0 (via `zmalloc` call, which calls `calloc` underneath).